### ISSUE TITLE
Allow to reopen closed webviews

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#97285]
Webviews appear to not restore via `Cmd+Shift+T` or running the `Reopen Closed Editor` command:

1. Open a webview, like a PR/welcome page/release notes
2. Close tab
3. Restore tab
4. :bug: tab is skipped and goes to next

This is also affected by pinned tabs but I assume it's related to the reopen command.

**Comment by bpasero** [comment]
This is intentional because there is no way currently a non-file based editor can signal that it can simply be restored. Check is here: 

https://github.com/microsoft/vscode/blob/602d2a217c6fd4c25afc14467ead4eae0c7923ce/src/vs/workbench/services/history/browser/history.ts#L615

However this could possibly be improved by simply asking the editor input factory to serialize the editor and then bring it back, I can play around with that approach.

@mjbvz not sure that makes sense to you, is it safe to serialize any webview via its factory and bring it back from that state?

**Comment by bpasero** [comment]
Upon second thought, I think my proposed idea is also not good because when an editor closes, assuming that you can serialize it may be risky. I think we would need some new infrastructure to support this scenario.

**Comment by bpasero** [comment]
@mjbvz with the change in https://github.com/microsoft/vscode/commit/1eac96d162f7ab7792e271e6dad12cf1ef31a29b, we are now asking the editor input factory to serialise an editor when it closes to support for opening it again. However it looks like webviews are not being serialized when they got closed, which maybe is for a good reason. 

If we cannot support this easily, then I suggest to close this feature, otherwise please try it out and see what is missing.

**Comment by rlabrecque** [comment]
Just want to throw my two cents in/use case in.

I came here because I updated to 1.47.0-insider from 1.46.0-insider, the initial launch opened the Release Notes: 1.47.0-insider tab, I closed it by accident, then tried to open it again with Ctrl+Shift+T.
Presumably that page is a webview and suffers from this.

Easy enough to get it back via Help -> Release Notes, but the expected actions did not work.
